pyproject.toml

```diff
@@ -93,8 +92,27 @@ ignore = [
     "PLR2004", # Magic value used in comparison, ...
     "S311", # Standard pseudo-random generators are not suitable for cryptographic purposes
     "SLF001", # OK to access private members in unit tests!
+    "W291", # Test files may have trailing whitespace intentionally
+    "W293", # Test files may have blank lines with whitespace intentionally
+]
+
+"pipeline/core/watcher.py" = [
+    "SLF001",  # OK to access private builder methods
```

Comment:
SLF001 for watcher doesn't belong in the pyproject.toml file. We'd want to inline this the actual file (so it's visible), but also this particular one is probably incorrect code. (it's not OK to access private builder methods in the watcher)